### PR TITLE
Fix regression of fetch headers (close #2334)

### LINKED ISSUE #2334: Authorization Header gets removed when passing request to server
### What is your Test Scenario?

When making a call to a backend api with authorization header, the call will go untouched to the backend server. Note this problem only started happening in version 1.8.5 (1.8.4 is working fine)

### What is the Current behavior?

Example of curl req going through testcafe to httpbin.org
Note the authorization header does not make it to httpbin.org

```
 curl 'http://<localhost ip>:49748/lRd4kqmG3/https://httpbin.org/headers' \
  -H 'Connection: keep-alive' \
  -H 'accept: application/json, text/plain, */*' \
  -H 'authorization: Bearer ya29.xxxxxxxxxxxxxx' \
  -H 'content-type: application/json;charset=UTF-8' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4156.0 Safari/537.36' \
  -H 'x-hammerhead-origin: https://my.website.com' \
  -H 'x-hammerhead-credentials: same-origin' \
  -H 'Referer: http://<localhost ip>:64928/HmgivWWBJ/https://my.website.com/' \
  -H 'Accept-Language: en-GB,en-US;q=0.9,en;q=0.8' \
  --compressed \
  --insecure
{
  "headers": {
    "Accept": "application/json, text/plain, */*",
    "Accept-Encoding": "deflate, gzip",
    "Accept-Language": "en-GB,en-US;q=0.9,en;q=0.8",
    "Content-Type": "application/json;charset=UTF-8",
    "Host": "httpbin.org",
    "Origin": "https://my.website.com",
    "Referer": "https://my.website.com/",
    "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4156.0 Safari/537.36",
    "X-Amzn-Trace-Id": "Root=1-5ecc53fb-b2c915fe220a7580f83xxxxx"
  }
}
```
### What is the Expected behavior?

Expect the request to pass to backend server intact with all headers. 

### What is your web application and your TestCafe test code?

 
<details>
<summary>Your website URL (or attach your complete example):</summary>

</details>

<details>
<summary>Your complete test code (or attach your test files):</summary>

 
```js
 
```
</details>

<details>
<summary>Your complete configuration file (if any):</summary>

```

```
</details>

<details>
<summary>Your complete test report:</summary>

```

```
</details>

<details>
<summary>Screenshots:</summary>

```

```
</details>

### Steps to Reproduce:

see curl above
 
### Your Environment details:
 
* testcafe version:  1.8.5                 
* node.js version:    10.16.3 / 12.16.1                
* command-line arguments:  testcafe chromium:headless tests/** -s path=reports,takeOnFails=true --reporter html:reports/report.html  --skip-js-errors

* browser name and version: Chromium 85.0.4156.0
* platform and version:          macOS Version 10.15.4
* other:

### PR DESCRIPTION
**Pull request by LavrovArtem**

### CODE REVIEW

File: src/client/sandbox/fetch.ts [hunk 1]
```diff
@@ -244,8 +234,44 @@ export default class FetchSandbox extends SandboxBase {
...
+            if (!result) {
+                args[0] = headerName;
+
+                return nativeMethods.headersHas.apply(this, args);
+            }
+
+            return result;
+        });
+
+        overrideFunction(window.Headers.prototype, 'has', function (...args) {
```
Comment (DenyVeyten) [inline src/client/sandbox/fetch.ts:269]: Shouldn't it be `'set'`?

Comment (miherlosev) [reply src/client/sandbox/fetch.ts:269]: Yes, it should be `set`. Also, add a test for the `set` method call.

**Review by miherlosev — CHANGES_REQUESTED** [review]

**Review by miherlosev — APPROVED** [review]

**Comment by LavrovArtem** [comment]
@testcafe-build-bot retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot retest

**Pull request merged**